src/btree/bt_split.c

```diff
@@ -1443,11 +1443,45 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
         if (!supd->restore)
             continue;
 
+        if (supd->ins == NULL) {
+            slot = WT_ROW_SLOT(orig, supd->ripcip);
+            upd = orig->modify->mod_row_update[slot];
+        } else
+            upd = supd->ins->upd;
+
+        /* We shouldn't restore an empty update chain. */
+        WT_ASSERT(session, upd != NULL);
+
+        /*
+         * Truncate the onpage value and the older versions moved to the history store. We can't
+         * truncate the updates for in memory database and fixed length column store as they don't
+         * support the history sore. We can't free the truncated updates here as we may still fail.
+         * If we fail, we will append them back to their original update chains. Truncate before we
+         * restore them to ensure the size of the page is correct.
+         */
+        if (supd->onpage_upd != NULL && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY) &&
+          orig->type != WT_PAGE_COL_FIX) {
+            /*
+             * We have decided to restore this update chain so it must have newer updates than the
+             * onpage value on it.
+             */
+            WT_ASSERT(session, upd != supd->onpage_upd);
+            /*
+             * Move the pointer to the position before the onpage value and free all the updates
+             * starting from the onpage value.
```

Comment:
Trivial point, but the updates aren't being "freed" here, so the comment is a little misleading.

Reply:
Fixed